### PR TITLE
Release Dafny 3.12.0

### PR DESCRIPTION
**Pull request by alex-chew**
<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+- Implements error detail information and quick fixes:
+     - An error catalog with error message explanations is at https://dafny.org/latest/HowToFAQ/Errors
+     - In VSCode, when hovering over an error, the hover information shows additional explanation and
+       an error id, which is also a link to the error explanation page
+     - Where a Quick Fix is available, the Quick Fix link is active
+  (https://github.com/dafny-lang/dafny/pull/3299)
+
+- * `opaque` is now a modifier, though still allowed, but deprecated as an identifier; it replaces the `{:opaque}` attribute (https://github.com/dafny-lang/dafny/pull/3462)
+
+- * The value of the --library option is allowed to be a comma-separated list of files or folders (https://github.com/dafny-lang/dafny/pull/3540)
```
Comment (atomb) [inline RELEASE_NOTES.md:18]: Isn't the formatter also a new feature?

**Review by atomb — CHANGES_REQUESTED** [review]
I think the formatter must have been missing a release notes file? I'm pretty sure it's new in this release.

File: RELEASE_NOTES.md (hunk 1 above)
Comment (alex-chew) [reply RELEASE_NOTES.md:22]: Good call, I've added a bullet point for this.

**Review by atomb — APPROVED** [review]
Looks good!

**Pull request merged**